### ISSUE TITLE
Investigate debugging of bats-core tests

### ISSUE THREAD

**Issue by jansorg** [BashSupport-Pro/bashsupport-pro#43]
https://github.com/BashSupport-Pro/bashsupport-pro/issues/42

Investigate about the best way to debug bats files.

**Comment by msandres13** [comment]
Looks like I have that problem now. Not sure if it was caused by one of the resent updates. bats debugging used to work great and now I cannot debug either all bats or single bats anymore.
I can run it, then it creates a configuration but debug symbol is disabled.
I can create debug data, let me know how.

**Comment by msandres13** [comment]
![image](https://user-images.githubusercontent.com/29023866/124917934-b49af300-dff4-11eb-86c1-e18db1e5139c.png)

**Comment by jansorg** [comment]
@msandres13 Thanks for reporting.
Could you post the version of BashSupport Pro and of your IDE, please? (e.g. `Help > About > Copy`)

Debugging of bats-core was never supported so far, at least not intentionally.
When you clicked "Debug" for a bats-core configuration, did it actually open the debugger and stop at breakpoints?

Adding breakpoints to .bats files shouldn't be possible, for example. The debug button should've been grayed out. But the version of the IDE might make a difference.

**Comment by msandres13** [comment]
> Debugging of bats-core was never supported so far, at least not intentionally.
> When you clicked "Debug" for a bats-core configuration, did it actually open the debugger and stop at breakpoints?
Yes, it did. I did set the breakpoint not in the bats, I set it in the code called in the testcase. I could not set a breakpoint in the bats itself nor run the debugger there.

**Comment by msandres13** [comment]
I also recognized that the bats icon in screen shoot I sent above is now grayed out.

**Comment by jansorg** [comment]
@msandres13 Thanks. You seem to have found a workaround to make the debugger run a bats test :) 
I'll see if I can reproduce this and if it's possible to enable it for all (again).

Because bats-core files are not syntactically correct bash, it wasn't enabled (intentially) yet.

The grayed out icon is telling that the run configuration isn't saved, i.e. it's a temporary configuration. That's the same treatment as with any other type of run configuration.

You're using a 2.0 preview. If you need debugging of bats, please try to uninstall the 2.0 preview and install the current stable version 1.5.20.211 again.

**Comment by msandres13** [comment]
Ok, I will try the workaround. I hope you can find a solution. Thanks for your help.

**Comment by rfay** [comment]
I would sure love to be able to debug bats tests.
